### PR TITLE
Fixes #YUPTOO-35 - Optimized Yuptoo code to use less memory

### PR DESCRIPTION
**Pull request by patilsuraj767**
Instead of loading all uuid.json files into single array, with this PR uuid.json files should be processed one by one. Hoping python garbage collector should be freeing up memory in each iteration of for loop - [here](https://github.com/patilsuraj767/yuptoo/blob/use-less-memory/yuptoo/processor/report_processor.py#L141).

### CODE REVIEW

File: yuptoo/processor/report_processor.py [hunk 1]
```diff
@@ -177,27 +178,14 @@ def extract_report_slices(report_tar, request_obj):
                                 continue
 
                             # Here performace can be improved by using Async thread
```
Comment (kgaikwad) [inline yuptoo/processor/report_processor.py:180]:
```suggestion
                            # Here performance can be improved by using Async thread
```

File: yuptoo/processor/report_processor.py [hunk 2]
```diff
@@ -177,27 +178,14 @@ def extract_report_slices(report_tar, request_obj):
...
-                            report_files.append(report_slice_json)
-                return report_files
+                            process_report_slice(report_slice_json, request_obj)
+                            del report_slice_json
```
Comment (kgaikwad) [inline yuptoo/processor/report_processor.py:182]: What is the intention here? Do you want to delete json object? GC does automatic deallocation, right?

Comment (patilsuraj767) [reply yuptoo/processor/report_processor.py:182]: As discussed internally, After first commit in this PR QE reported that Yuptoo is still consuming the same memory as earlier. So I tried to manually delete object and invoked the garbage collector, But then we found that PromQL query which we are looking is incorrect. 
For now I will revert this changes and will ask QE to test again.

**Review by san7ket — APPROVED** [review]
Verified

File: yuptoo/processor/report_processor.py [hunk 3]
```diff
@@ -21,65 +21,58 @@
 FAILURE_CONFIRM_STATUS = 'failure'
 
 
-def process_report(consumed_message, p, request_obj):
-    global producer
-    producer = p
-    report_tar = download_report(consumed_message)
-    report_json_files = extract_report_slices(report_tar, request_obj)
-
+def process_report_slice(report_slice, request_obj):
```
Comment (kgaikwad) [inline yuptoo/processor/report_processor.py:24]: @patilsuraj767  - As you are changing the existing report processing, could you please add description of what the method is and what it's used for?

File: yuptoo/processor/report_processor.py [hunk 4]
```diff
@@ -176,28 +176,13 @@ def extract_report_slices(report_tar, request_obj):
                                 LOG.warning(mismatch_message)
                                 continue
 
-                            # Here performace can be improved by using Async thread
-                            report_files.append(report_slice_json)
-                return report_files
+                            # Here performance can be improved by using Async thread
```
Comment (kgaikwad) [inline yuptoo/processor/report_processor.py:179]:
```suggestion
                            # FIXME - performance can be improved by using Async thread
```

**Review by kgaikwad — CHANGES_REQUESTED** [review]
Added two inline nitpick comments.

File: yuptoo/processor/report_processor.py (hunk 3 above)
Comment (patilsuraj767) [reply yuptoo/processor/report_processor.py:24]: Done

**Pull request merged**